### PR TITLE
User Dashboard

### PR DESCRIPTION
**Pull request by dborski**
## User Dashboard View ##
* Added new layout for dashboard so that current user name appears in nav bar
* Add six news articles displayed in center of page
* Add representatives who have been compared with already 

## Senator and HouseMember Models ##
* Add has_many relationship with each model's respective favorites table

## User Model ##
* Add `has_favorites?` instance method and spec
* Add `compared_house_members` instance method and spec
* Add `compared_senators` instance method and spec

## Rake ##
* Add `update_reps` rake task that updates all HouseMember and Senator objects in db with updated info from representatives endpoint
* Add `update_representatives` method to propublica _parser

### CODE REVIEW

File: app/services/propublica_parser.rb [hunk 1]
```diff
@@ -14,6 +14,22 @@ def parse_house_members
...
+
+    representative.all.each do |representative|
+      rep_data = reps_json.find.each { |rep| rep[:congress_id] == representative.congress_id }
+      rep_data[:last_updated] = rep_data[:last_updated].to_datetime
+      if rep_data.delete(:in_office)
+        representative.update(rep_data)
+      end
+    end 
+  end
+
```
Comment (dborski) [inline app/services/propublica_parser.rb:32]: This method updates both house_members and senators and is called from the new rake task

Comment (Rostammahabadi) [reply app/services/propublica_parser.rb:32]: This is awesome

File: app/controllers/dashboard_controller.rb [hunk 2]
```diff
@@ -1,11 +1,13 @@
 class DashboardController < ApplicationController
+  layout "dashboard_layout"
```
Comment (Rostammahabadi) [inline app/controllers/dashboard_controller.rb:2]: html.erb 👍

**Review by Rostammahabadi — APPROVED** [review]

**Review by AntonioJacksonII — APPROVED** [review]
Nice! This dashboard page looks great.

File: lib/tasks/update_reps.rake [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    model_objects = [HouseMember, Senator]
+
+    puts 'Updating house_members and senators'
+    parser = PropublicaParser.new
+
+    model_objects.each do |model|
+      parser.update_representatives(model)
+    end 
+
+    puts 'JSON update task complete'
```
Comment (arpariseau) [inline lib/tasks/update_reps.rake:16]: Looking pretty sweet. 👍

**Review by arpariseau — APPROVED** [review]

**Pull request merged**